components/password_reset_send_link/password_reset_send_link.tsx

```diff
@@ -1,41 +1,46 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
+import {ServerError} from 'mattermost-redux/types/errors';
 import {isEmail} from 'mattermost-redux/utils/helpers';
 
 import BackButton from 'components/common/back_button';
 import LocalizedInput from 'components/localized_input/localized_input';
 
 import {t} from 'utils/i18n.jsx';
 
-export default class PasswordResetSendLink extends React.PureComponent {
-    static propTypes = {
-        actions: PropTypes.shape({
-            sendPasswordResetEmail: PropTypes.func.isRequired,
-        }).isRequired,
+interface Props {
+    actions: {
+        sendPasswordResetEmail: (emal: string) => Promise<{data: any; error: ServerError}>;
     };
+}
+
+interface State {
+    error: React.ReactNode;
+    updateText: React.ReactNode;
+}
 
+export default class PasswordResetSendLink extends React.PureComponent<Props, State> {
     state = {
         error: null,
         updateText: null,
     };
-    resetForm = React.createRef();
-    emailInput = React.createRef();
+    resetForm = React.createRef<HTMLFormElement>();
+    emailInput = React.createRef<HTMLInputElement>();
 
-    handleSendLink = async (e) => {
+    handleSendLink = async (e: React.FormEvent) => {
         e.preventDefault();
 
-        const email = this.emailInput.current.value.trim().toLowerCase();
+        const email = this.emailInput.current!.value.trim().toLowerCase();
```

Comment:
```suggestion
        const email = this.emailInput.current?.value.trim().toLowerCase();
```
Minor:  I think `?` would be a bit more appropriate